src/RedHat.AspNetCore.Server.Kestrel.Transport.Linux/TransportThread.TSocket.cs

```diff
@@ -61,8 +59,13 @@ public struct ReceiveMemoryAllocation
             private const int EventControlRegistered = (int)SocketFlags.EventControlRegistered;
             public const int EventControlPending = (int)SocketFlags.EventControlPending;
 
-            private static readonly int MaxBufferSize = KestrelMemoryPool.MinimumSegmentSize;
-            private static readonly int BufferMargin = MaxBufferSize / 4;
+            // Copied from LibuvTransportOptions.MaxReadBufferSize
+            private static readonly int PauseInputWriterThreashold = 1024 * 1024;
+            // Copied from LibuvTransportOptions.MaxWriteBufferSize
+            private static readonly int PauseOutputWriterThreashold = 64 * 1024;
+
+            private readonly int MaxBufferSize;
+            private readonly int BufferMargin;
```

Comment:
We don't need to store these per `TSocket`. You can make them properties, that get the values from `MemoryPool`.